packages/tools/src/tools/base/AnnotationTool.ts

```diff
@@ -56,10 +57,12 @@ abstract class AnnotationTool extends AnnotationDisplayTool {
    *
    * @param evt - The normalized mouse event
    * @param interactionType -  The interaction type used to add the annotation.
+   * @param toolBinding - Tool binding information
    */
   abstract addNewAnnotation(
     evt: EventTypes.InteractionEventType,
-    interactionType: InteractionTypes
+    interactionType: InteractionTypes,
+    toolBinding: IToolBinding
```

Comment:
There are two binding for the same tool where `click` would just draw an annotation and `shift+click` would draw and append/remove/create hole if it intersects other contours. Then we need to know which binding was used to draw annotations and instead of using `shift` modifier only (hardcoded) I added data that allow us to use any binding as the activator. Just need to use `data` to add an extra information for an specific binding to turn that on/off.

Reply:
Code removed after pushing a new way to detect modifiers keys